echopype/tests/consolidate/test_consolidate_integration.py

```diff
@@ -147,36 +151,430 @@ def _build_ds_Sv(channel, range_sample, ping_time, sample_interval):
+
+    # Compute `depth` using platform vertical offset values
+    ds_Sv = ep.consolidate.add_depth(ds_Sv, ed, use_platform_vertical_offsets=True)
+
+    # Check history attribute
+    history_attribute = ds_Sv["depth"].attrs["history"]
+    history_attribute_without_time = history_attribute[33:]
+    assert history_attribute_without_time == (
+        ". depth` calculated using: Sv `echo_range`, Echodata `Platform` Vertical Offset Data."
+    )
+
+    # Compute transducer depth
+    transducer_depth = ek_use_platform_vertical_offsets(ed["Platform"], ds_Sv["ping_time"])
+    
+    # Iterate through every channel
+    for channel_index in range(len(ds_Sv["channel"])):
+        # Iterate through every ping time
+        for ping_time_index in range(len(ds_Sv["ping_time"])):
+            # Iterate through every range sample
+            for range_sample_index in range(len(ds_Sv["range_sample"])):
+                # Extract relevant values
+                depth_value = ds_Sv["depth"].isel(
+                    channel=channel_index, ping_time=ping_time_index, range_sample=range_sample_index
+                ).data
+                transducer_depth_value = transducer_depth.isel(
+                    channel=channel_index, ping_time=ping_time_index
+                ).data
+                # Check if depth value is equal to corresponding `echo_range` value + transducer depth value
+                assert np.isclose(
+                    depth_value,
+                    ds_Sv["echo_range"].isel(
+                        channel=channel_index, ping_time=ping_time_index, range_sample=range_sample_index
+                    ).data + transducer_depth_value,
+                    rtol=1e-10,
+                    atol=1e-10
+                )
+
+
+@pytest.mark.integration
+@pytest.mark.parametrize("file, sonar_model, compute_Sv_kwaargs", [
+    (
+        "echopype/test_data/ek60/NBP_B050N-D20180118-T090228.raw",
+        "EK60",
+        {}
+    ),
+    (
+        "echopype/test_data/ek60/ncei-wcsd/Summer2017-D20170620-T021537.raw",
+        "EK60",
+        {}
+    ),
+    (
+        "echopype/test_data/ek80/ncei-wcsd/SH1707/Reduced_D20170826-T205615.raw",
+        "EK80",
+        {"waveform_mode":"BB", "encode_mode":"complex"}
+    ),
+    (
+        "echopype/test_data/ek80/ncei-wcsd/SH2106/EK80/Reduced_Hake-D20210701-T131621.raw",
+        "EK80",
+        {"waveform_mode":"CW", "encode_mode":"power"}
+    )
+])
+def test_add_depth_EK_with_platform_angles(file, sonar_model, compute_Sv_kwaargs):
+    """Test `depth` values when using EK Platform angles to compute it."""
+    # Open EK Raw file and Compute Sv
+    ed = ep.open_raw(file, sonar_model=sonar_model)
+    ds_Sv = ep.calibrate.compute_Sv(ed, **compute_Sv_kwaargs)
+
+    # Replace any Beam Angle NaN values with 0
+    ed["Platform"]["pitch"] = ed["Platform"]["pitch"].fillna(0)
+    ed["Platform"]["roll"] = ed["Platform"]["roll"].fillna(0)
```

Comment:
Are these just adding them so that the add_depth computation would go through?

Reply:
Yeah, else the values would just be NaN.